packages/xo-web/src/common/xo/add-hosts-modal/index.js

```diff
@@ -50,57 +50,78 @@ export default class AddHostsModal extends BaseComponent {
     ) {
       return {}
     }
-
-    return { hosts: this.state.hosts }
+    const { host, hosts } = this.state
+    return { hosts: host === undefined ? hosts : [host] }
   }
 
   _getHostPredicate = createSelector(
     () => this.props.singleHosts,
     singleHosts => host => singleHosts[host.id]
   )
 
-  _onChangeHosts = async hosts => {
-    if (isEmpty(hosts)) {
+  _onChangeHost = async value => {
+    if (isEmpty(value)) {
       this.setState({
-        hosts,
+        host: undefined,
+        hosts: undefined,
         nHostsMissingPatches: undefined,
         nPoolMissingPatches: undefined,
       })
       return
     }
-
     const { master } = this.props.pool
-    this.setState({
-      hosts,
-      nHostsMissingPatches: uniq(
-        flatten(
-          await Promise.all(
-            map(hosts, ({ id: hostId }) => getPatchesDifference(hostId, master))
+    if (+process.env.XOA_PLAN === 4) {
+      this.setState({
+        hosts: value,
+        nHostsMissingPatches: uniq(
+          flatten(
+            await Promise.all(
+              map(value, ({ id: hostId }) =>
+                getPatchesDifference(hostId, master)
+              )
+            )
           )
-        )
-      ).length,
-      nPoolMissingPatches: uniq(
-        flatten(
-          await Promise.all(
-            map(hosts, ({ id: hostId }) => getPatchesDifference(master, hostId))
+        ).length,
+        nPoolMissingPatches: uniq(
+          flatten(
+            await Promise.all(
+              map(value, ({ id: hostId }) =>
+                getPatchesDifference(master, hostId)
+              )
+            )
           )
-        )
-      ).length,
-    })
+        ).length,
+      })
+    } else {
+      this.setState({
+        host: value,
+        nHostsMissingPatches: (await getPatchesDifference(value.id, master))
+          .length,
+        nPoolMissingPatches: (await getPatchesDifference(master, value.id))
+          .length,
+      })
+    }
   }
 
   render() {
-    const { hosts, nHostsMissingPatches, nPoolMissingPatches } = this.state
+    const {
+      host,
+      hosts,
+      nHostsMissingPatches,
+      nPoolMissingPatches,
+    } = this.state
     return (
       <div>
         <SingleLineRow>
-          <Col size={6}>{_('hosts')}</Col>
+          <Col size={6}>
+            {+process.env.XOA_PLAN === 4 ? _('hosts') : _('homeTypeHost')}
+          </Col>
           <Col size={6}>
             <SelectHost
-              multi
-              onChange={this._onChangeHosts}
+              multi={+process.env.XOA_PLAN === 4}
+              onChange={this._onChangeHost}
               predicate={this._getHostPredicate()}
-              value={hosts}
+              value={+process.env.XOA_PLAN === 4 ? hosts : host}
```

Comment:
Let's handle everything as if the user could select multiple hosts and add a simple change like:
```js
<SelectHost
  multi={canMulti}
  onChange={canMulti ? this._onChangeHosts : host => this._onChangeHosts([host])}
  value={canMulti ? hosts : hosts[0]}
/>
```